src/mpi/init/init_async.c

```diff
@@ -166,12 +168,11 @@ int MPIR_Finalize_async_thread(void)
     MPID_Thread_mutex_unlock(&progress_mutex, &mpi_errno);
     MPIR_Assert(!mpi_errno);
 
+    MPID_Thread_join(progress_thread_id);
+
     mpi_errno = MPIR_Comm_free_impl(progress_comm_ptr);
```

Comment:
Should `MPIR_Comm_free_impl` here be protected?

Reply:
Can you try testing with `--enable-g` to confirm if this comm free call needs to be protected? I would see occasional handle leaks because of an un-freed communicator in my local testing. 3.3.x branch also has this issue.

Reply:
sure.

Reply:
Tests passed, so good enough for me.